tests/gc.rs

```diff
@@ -0,0 +1,521 @@
+        .set_tag(tag.clone(), Some(HashAndFormat::raw(h2)))
+        .await?;
+    drop(tt2);
+    tracing::info!("dropped tt2");
+    step(&evs).await;
+    assert_eq!(bao_store.entry_status(&h2).await?, EntryStatus::Complete);
+
+    // delete the explicit tag, entry should be gone
+    bao_store.set_tag(tag, None).await?;
+    step(&evs).await;
+    assert_eq!(bao_store.entry_status(&h2).await?, EntryStatus::NotFound);
+
+    node.shutdown().await?;
+    Ok(())
+}
+
+/// Test gc for sequences of hashes that protect their children from deletion.
+#[tokio::test]
+async fn gc_hashseq_impl() -> Result<()> {
+    let _ = tracing_subscriber::fmt::try_init();
+    let (node, bao_store, evs) = gc_test_node().await;
+    let data1 = create_test_data(1234);
+    let tt1 = bao_store.import_bytes(data1, BlobFormat::Raw).await?;
+    let data2 = create_test_data(5678);
+    let tt2 = bao_store.import_bytes(data2, BlobFormat::Raw).await?;
+    let seq = vec![*tt1.hash(), *tt2.hash()]
+        .into_iter()
+        .collect::<HashSeq>();
+    let ttr = bao_store
+        .import_bytes(seq.into_inner(), BlobFormat::HashSeq)
+        .await?;
+    let h1 = *tt1.hash();
+    let h2 = *tt2.hash();
+    let hr = *ttr.hash();
+    drop(tt1);
+    drop(tt2);
+
+    // there is a temp tag for the link seq, so it and its entries should be there
+    step(&evs).await;
+    assert_eq!(bao_store.entry_status(&h1).await?, EntryStatus::Complete);
+    assert_eq!(bao_store.entry_status(&h2).await?, EntryStatus::Complete);
+    assert_eq!(bao_store.entry_status(&hr).await?, EntryStatus::Complete);
+
+    // make a permanent tag for the link seq, then delete the temp tag. Entries should still be there.
+    let tag = Tag::from("test");
+    bao_store
+        .set_tag(tag.clone(), Some(HashAndFormat::hash_seq(hr)))
+        .await?;
+    drop(ttr);
+    step(&evs).await;
+    assert_eq!(bao_store.entry_status(&h1).await?, EntryStatus::Complete);
+    assert_eq!(bao_store.entry_status(&h2).await?, EntryStatus::Complete);
+    assert_eq!(bao_store.entry_status(&hr).await?, EntryStatus::Complete);
+
+    // change the permanent tag to be just for the linkseq itself as a blob. Only the linkseq should be there, not the entries.
+    bao_store
+        .set_tag(tag.clone(), Some(HashAndFormat::raw(hr)))
+        .await?;
+    step(&evs).await;
+    assert_eq!(bao_store.entry_status(&h1).await?, EntryStatus::NotFound);
+    assert_eq!(bao_store.entry_status(&h2).await?, EntryStatus::NotFound);
+    assert_eq!(bao_store.entry_status(&hr).await?, EntryStatus::Complete);
+
+    // delete the permanent tag, everything should be gone
+    bao_store.set_tag(tag, None).await?;
+    step(&evs).await;
+    assert_eq!(bao_store.entry_status(&h1).await?, EntryStatus::NotFound);
+    assert_eq!(bao_store.entry_status(&h2).await?, EntryStatus::NotFound);
+    assert_eq!(bao_store.entry_status(&hr).await?, EntryStatus::NotFound);
+
+    node.shutdown().await?;
+    Ok(())
+}
+
+mod file {
```

Comment:
does this need a cfg?

Reply:
Somehow the feature flag for the fs store got lost. Not sure if we need it. The only thing fs-store brings in in terms of deps is redb, and that is very lightweight.